h2o-algos/src/main/java/hex/word2vec/Word2VecModel.java

```diff
@@ -41,6 +41,46 @@ public Word2VecMojoWriter getMojo() {
   }
 
   /**
+   * Converts this word2vec model to a Frame.
+   * @return Frame made of columns: Word, V1, .., Vn. Word column holds the vocabulary associated
+   * with this word2vec model, and columns V1, .., Vn represent the embeddings in the n-dimensional space.
+   */
+  public Frame toFrame() {
+    Vec zeroVec = null;
+    try {
+      zeroVec = Vec.makeZero(_output._words.length);
+      byte[] types = new byte[1 + _output._vecSize];
+      Arrays.fill(types, Vec.T_NUM);
+      types[0] = Vec.T_STR;
+      String[] colNames = new String[types.length];
+      colNames[0] = "Word";
+      for (int i = 1; i < colNames.length; i++)
+        colNames[i] = "V" + i;
+      return new ConvertToFrameTask(this).doAll(types, zeroVec).outputFrame(colNames, null);
+    } finally {
+      if (zeroVec != null) zeroVec.remove();
+    }
+  }
+
+  private static class ConvertToFrameTask extends MRTask<ConvertToFrameTask> {
+    private Word2VecModel _model;
```

Comment:
The model `key` should be enough no? and then pre-fetch the model in `setupLocal`.
(Note: i am not sure now, if there is optimization in serialization layer which replaces models by its key).

Furthermore, if you decide to go with Model then we can replace `_model` reference by `null` at map call (to avoid transfer of model back).